table/table.go

```diff
@@ -266,7 +266,11 @@ func (m *Model) UpdateViewport() {
 // SelectedRow returns the selected row.
 // You can cast it to your own implementation.
 func (m Model) SelectedRow() Row {
-	return m.rows[m.cursor]
+	// Check if the cursor is within bounds else return  nil
+	if m.cursor < len(m.rows) && m.cursor >= 0 {
```

Comment:
May I suggest swapping the logic around? We should check for `m.cursor` being out of bounds as the exception and keep the "happy path" on the left edge.

Reply:
You mean? 
```suggestion
	if m.cursor >= 0 && m.cursor < len(m.rows) {
```

Reply:
Great suggestion.